include/real/exact_number.hpp

```diff
@@ -693,190 +694,322 @@ namespace boost {
                 remainder = exact_remainder.digits;
             }
 
-            /// calculates *this / divisor 
-            /// 
-            ///  @brief a binary-search type method for dividing exact_numbers.
-            ///  @param is_upper true: returns result with an error of +epsilon, while
-            ///                  false: returns result with an error of -epsilon
-            void divide_vector(exact_number<T> divisor, unsigned int maximum_precision) {
-                /// @TODO: replace this with something more efficient, like newton-raphson method
-                // it also completely recalculates on each precision increase
-                // instead, could use previous information to make better "guesses"
-                // for our iteration scheme.
-
-                boost::real::exact_number<T> numerator;
-                boost::real::exact_number<T> left;
-                boost::real::exact_number<T> right;
-                boost::real::exact_number<T> residual;
-                boost::real::exact_number<T> tmp;
-                boost::real::exact_number<T> half;
-                boost::real::exact_number<T> distance;
-                boost::real::exact_number<T> min_boundary_n;
-                boost::real::exact_number<T> min_boundary_p;
-
-                bool positive = ((*this).positive == divisor.positive);
-                numerator = (*this).abs();
-                divisor = divisor.abs();
-
-                // ensuring that assignment from -1 * (maximum_precision) to exponent will not
-                // overflow
+            /*              DIVISION
+             *  @brief:  divides (*this) by divisor
+             *  @param:  divisor: number which divides (*this)
+             *  @param:  maximum_precision: Absolute Error in the result should be < 1*base^(-maximum_precision)
+             *  @param:  upper: if true: error lies in [0, +epsilon]
+             *                  else: error lies in [-epsilon, 0], here epsilon = 1*base^(-maximum_precision)
+             *  @author: Kishan Shukla
+             */
+            void divide_vector(const exact_number<T> divisor, unsigned int maximum_precision, bool upper) {
+
+                newton_raphson_division(divisor, maximum_precision, upper);
+
+            }
+
+            /*          BINARY SEARCH BASED DIVISION
+             *      an approximate division method, used in the initial guess of reciprocal in Newton Raphson
+             */
+
+            void binary_search_division(
+                const exact_number<T> divisor,
```

Comment:
Do we have same function implemented for divisors that are not exact_numbers?

Reply:
No, because we perform division only on the boundaries of the interval and boundaries are always exact_numbers. Am I wrong?

Reply:
No, that is correct, we only need divisors for the boundaries (exact numbers)